README.md

```diff
@@ -3,30 +3,233 @@
+    - [Default retry](#setup-client-default-retry)
+    - [Logger](#setup-client-logger)
+  - [Producing Messages to Kafka](#producing-messages)
+    - [Custom partitioner](#producing-messages-custom-partitioner)
+    - [Retry](#producing-messages-retry)
+  - [Consuming messages from Kafka](consuming-messages)
+    - [Options](consuming-messages-options)
+    - [Custom assigner](#consuming-messages-custom-assigner)
+  - [Instrumentation](#instrumentation)
+  - [Development](#development)
+
+## <a name="installation"></a> Installation
+
+```sh
+npm install kafkajs
+# yarn add kafkajs
+```
+
+## <a name="usage"></a> Usage
+
+### <a name="setup-client"></a> Setting up the Client
 
-### Setting up the Client
+The client must be configured with at least one broker. The brokers on the list are considered seed brokers and are only used to bootstrap the client and load initial metadata.
 
 ```javascript
 const { Kafka } = require('kafkajs')
 
-// Create the client with broker list
+// Create the client with the broker list
 const kafka = new Kafka({
   clientId: 'my-app',
   brokers: ['kafka1:9092', 'kafka2:9092']
 })
 ```
 
-### Producing Messages to Kafka
+#### <a name="setup-client-ssl"></a> SSL
+
+The `ssl` option can be used to configure the TLS sockets. The options are passed directly to `tls.connect` and used to create the TLS Secure Context, all options are accepted.
+
+```javascript
+const fs = require('fs')
+
+new Kafka({
+  clientId: 'my-app',
+  brokers: ['kafka1:9092', 'kafka2:9092'],
+  ssl: {
+    rejectUnauthorized: false,
+    ca: [fs.readFileSync('/my/custom/ca.crt', 'utf-8')],
+    key: fs.readFileSync('/my/custom/client-key.pem', 'utf-8'),
+    cert: fs.readFileSync('/my/custom/client-cert.pem', 'utf-8')
+  },
+})
+```
+
+Take a look at [TLS create secure context](https://nodejs.org/dist/latest-v8.x/docs/api/tls.html#tls_tls_createsecurecontext_options) for more information.
+
+#### <a name="setup-client-sasl"></a> SASL
+
+Kafka has support for using SASL to authenticate clients. The `sasl` option can be used to configure the authentication mechanism. Currently, KafkaJS only supports the `PLAIN` mechanism.
+
+```javascript
+new Kafka({
+  clientId: 'my-app',
+  brokers: ['kafka1:9092', 'kafka2:9092'],
+  sasl: {
+    mechanism: 'plain',
+    username: 'my-username',
+    password: 'my-password'
+  },
+})
+```
+
+It is __highly recommended__ that you use SSL for encryption when using `PLAIN`.
+
+#### <a name="setup-client-connection-timeout"></a> Connection Timeout
+
+Time in milliseconds to wait for a successful connection. The default value is: `1000`.
+
+```javascript
+new Kafka({
+  clientId: 'my-app',
+  brokers: ['kafka1:9092', 'kafka2:9092'],
+  connectionTimeout: 3000
+})
+```
+
+#### <a name="setup-client-default-retry"></a> Default Retry
+
+The `retry` option can be used to set the default configuration. The retry mechanism uses a randomization function that grows exponentially.
```

Comment:
You could mention what you actually retry: Just the connect or also the production of messages or ...?
Also: Mention what happens after the max number of retries has exceeded.

Reply:
done